src/braket/experimental/autoqasm/converters/return_statements.py

```diff
@@ -0,0 +1,40 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+"""Converters for return statement nodes."""
+
+import ast
+import warnings
+
+from braket.experimental.autoqasm import program
+from braket.experimental.autoqasm.autograph.converters import return_statements
+from braket.experimental.autoqasm.autograph.core import ag_ctx, converter
+
+
+class ReturnValidator(converter.Base):
+    def visit_Return(self, node: ast.stmt) -> ast.stmt:
+        aq_context = program.get_program_conversion_context()
+        if aq_context.is_in_main and node.value is not None:
+            warnings.warn("`output` is currently unsupported; `return` statement has no effect.")
```

Comment:
Technically the `return` statement still has an effect (it ends execution of the program), it's only the returned object that is ignored. Also, will the user know what `output` means in this context? (that's an OpenQASM concept)

Reply:
Good points -- Struggling to think of a way to explain to the user that they might not want to return a value without connecting to openqasm. Maybe "Returning a value from the top level function call has undefined behavior"? Or, we could close this and allow it to be ambiguous until we implement `output`

Reply:
Yeah this is a weird one to explain - especially if we plan to support it in the future. Maybe just something to communicate that "returning a value from the top-level function doesn't do anything right now".